#8 combat

jetenergy opened the pull request.

zoompower committed:
First version combat

zoompower committed:
Combat mind

zoompower committed:
Merge branch 'Develop' into #8-Combat

zoompower committed:
Update CombatMind.cs

zoompower committed:
Update CombatMind.cs

zoompower committed:
Update CombatMind.cs

zoompower committed:
Update CombatMind.cs

zoompower committed:
Pushy

MMCQ33 committed:
new combatmind ui

Added UI in mindbuilder for new combat AI.

jetenergy committed:
Merge branch 'Develop' into #8-Combat

# Conflicts:
#	Hivemind/Assets/Scripts/Ant.cs
#	Hivemind/Assets/Scripts/Minds/CombatMind.cs
#	Hivemind/Assets/Scripts/UI/MindBuilderTabbed.cs

zoompower committed:
Poging 1

jetenergy committed:
Added null checking in gameresource
fixed hard coded mind giving
fixed ant not having correct teamId when getting controllers

jetenergy committed:
Gave ant the correct teamColor, input is on BaseController

zoompower committed:
Ants can now attack.

zoompower committed:
Merge branch 'teamColor-fix' into #8-Combat

zoompower committed:
Update Map.unity

jetenergy committed:
Fixed spatial partitioning memory leak
fixed error when target is dead not by my hands

zoompower committed:
Update SpatialPartitioning.cs

zoompower committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

zoompower committed:
Finishing combat mind and adding player 2 to map.Unity

zoompower committed:
Update Ant.cs

zoompower committed:
Update CombatMind.cs

MMCQ33 committed:
Added victory screen

zoompower committed:
Allow queen murder in cold blood

zoompower committed:
Merge branch 'VictoryScee(ch)n' into #8-Combat

zoompower committed:
Game can now be won/lost

jetenergy committed:
Merge branch 'Develop' into SpatialPartitionFix-MindBuilderFix

# Conflicts:
#	Hivemind/Assets/Resources/Prefabs/UI/MindBuilderTabbed.prefab
#	Hivemind/Assets/Scenes/Level_1.unity
#	Hivemind/Assets/Scenes/Map.unity
#	Hivemind/Assets/Scripts/Ant.cs
#	Hivemind/Assets/Scripts/UI/MindBuilderTabbed.cs

jetenergy committed:
forgot to save again

MMCQ33 committed:
fixed ant walking correctly

jetenergy committed:
Added base spatial partitioning

jetenergy committed:
Fixed death incorrect spatial removal
Cleaned combat
Fixed Incorrect dispaly of mind combat in inspector
Fixed mindbuilder test

zoompower committed:
Merge branch 'SpatialPartitionFix-MindBuilderFix' into #8-Combat

MMCQ33 committed:
Fixed level1 with new combat

fixed level1 with combat, added more food, added more resources to map, updated navmesh, added enemy base to level1, fixed some minor bugs.

MMCQ33 committed:
fixed ants getting resources immidieatly outside the base

jetenergy committed:
Fixed basegroup not being able to walk

MMCQ33 committed:
fixed map loading bug

MMCQ33 committed:
Merge remote-tracking branch 'origin/potential-combat-fix' into Level1-EnemyBase

MMCQ33 committed:
made enemybase just gather resources

MMCQ33 committed:
made ants not look trhough all partitions

ErikLouwerse committed:
General automated cleanup

ErikLouwerse committed:
Merge pull request #35 from zoompower/Code-Cleanup

General automated cleanup

MMCQ33 committed:
fixed ants not spawning correctly

ErikLouwerse committed:
Merge pull request #37 from zoompower/Code-Cleanup

fixed ants not spawning correctly

MMCQ33 committed:
Merge branch '#8-Combat' into CombatEnemyFix

MMCQ33 committed:
Fixed enemy ants not going to base

Enemy ants are now going to their base.

MMCQ33 committed:
Fixed nullreference in enemy base

MMCQ33 committed:
fixed ants not returning to field or base

Ants move now correctly towards base when they were attacked and moving to base previously.

MMCQ33 committed:
If ant is in combat just before entering base

MMCQ33 committed:
FIxed navmesh

Made voxel size smaller to fix navmesh problem

MMCQ33 committed:
Ants do not return to base after killing an enemy

MMCQ33 committed:
removed notimplementedfunction

zoompower committed:
Max ground fix

MMCQ33 committed:
Updated scaling for enemy

MMCQ33 committed:
Merge remote-tracking branch 'origin/MapFix' into #8-Combat

MMCQ33 committed:
Updated quit being highlighted when play was pressed

MMCQ33 committed:
Fixed ant moving to resource correctly

when changed to gathering ants are going to a resource correctly.

jetenergy committed:
Changed mindgroup minds to have getter and setter
Fixed likelyhood of basegroupmind being too high
Fixed combatmind incorrect state after attacking enemy while attacking queen and not in enemy base
Updated tests

jetenergy committed:
Fixed visual mind overrides

MMCQ33 committed:
Made Bases bigger

Made bases twice their size

MMCQ33 committed:
Updated acceleration to work better with new speed

MMCQ33 committed:
Merge remote-tracking branch 'origin/Ai' into #8-Combat

MMCQ33 committed:
Added basic AI in level 1 and Changed some speeds around

Modified speed so it works correctly now.
Added Ai to the enemy.

ErikLouwerse committed:
Fixed saving and loading :)

ErikLouwerse committed:
merge

ErikLouwerse committed:
Added data for AI

jetenergy committed:
Different fix for spawn units when game loaded

ErikLouwerse committed:
Fixed error after returning to main menu and loading

ErikLouwerse committed:
Removed unitcontrollers when scene is unloaded to prevent increasing list

ErikLouwerse committed:
Fixed minimaprender scale

MMCQ33 committed:
small changes

Changed speed of ants when they returned to base and  delivering resources and added navmeshscripts to gitignore

jetenergy committed:
Fixed basebuilding loading issues

ErikLouwerse committed:
Changed basicAI in gameworld to be a list

ErikLouwerse committed:
PR comments

ErikLouwerse committed:
Pr comments

MMCQ33 committed:
Ants now return to queen after done with gathering

Ants now return to queen instead of standing on the teleporter

ErikLouwerse committed:
Merge branch '#8-Combat' into SaveLoadFix

jetenergy committed:
Merge pull request #40 from zoompower/SaveLoadFix

Save load fix

MMCQ33 committed:
Improved ant nav

Ants go to nest easier

MMCQ33 committed:
fixed ants scouting in base

fixed ants sometimes scouting in base and fixed some spacing

ErikLouwerse committed:
Changed save selection

MMCQ33 committed:
Ants attack each other when moving towards queen

Ants now attack enemy ants when moving towards the queen again

MMCQ33 committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

MMCQ33 committed:
readded ai

MMCQ33 committed:
Made minimap a bit more accurate

ErikLouwerse committed:
Optimized scouting
Changed Unknown resource to None
Changed Preffered Type to Exclusive Type

ErikLouwerse committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

jetenergy commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+
+Hivemind/Assets/Tests/PlayModeTests/NavMeshScripts.meta
```
Please remove this file.

ErikLouwerse committed:
further enhanced scouting so they can get to the other side of the map

jetenergy committed:
Changed BasicAi

Moved priorities arround,
fixed saving of basicAi

MMCQ33 committed:
balanced food

MMCQ33 committed:
Udated enemy pathing

Enemies dont get stuck outside of their base anymore

MMCQ33 committed:
check if entitylist isnt null

MMCQ33 committed:
fixed food being nerfed

ErikLouwerse committed:
Ants return to queen after combat and fixed ai loading

MMCQ33 committed:
Updated rocks, made code cleaner

Made code cleaner, removed some fields in combatmind, renamed basecontroller in basicAi, and moved rocks so ants dont get stuck so often

MMCQ33 committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

jetenergy committed:
Added 50 unit limit to basebuilding and enemy Ai
Fixed splitting room counts

MMCQ33 committed:
Updated Tree branches

Ants can now walk onto the middle plateau.

MMCQ33 committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

MMCQ33 committed:
fixed some objects being walked through

fixed some objects not being static meaning ants would walk through them

ErikLouwerse committed:
Updating workerroom remaining WIP

MMCQ33 committed:
removed unneeded data

removed fields and their references

MMCQ33 committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

ErikLouwerse committed:
Workerroomamount WIP

ErikLouwerse committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

MMCQ33 committed:
Removed gitignore, cleaned code for roombutton

Removed an unneeded second gitignore, Cleaned debug code., fixed ant room button now displaying correct amount of rooms left.

MMCQ33 committed:
fix not being consisten

fix button not giving correct amount sometimes.

jetenergy committed:
Changed variable name

MMCQ33 commented on Hivemind/Assets/Scripts/Ant.cs:
```diff
@@ -130,6 +140,34 @@ public bool InCombat()
+    public void SetClosestEnemy(Ant c)
+    {
+        closestEnemy = c;
+    }
```
C can be more clear: like cEnemy or closestEnemy

ErikLouwerse committed:
Change button interactable based on how many unitrooms you have

ErikLouwerse committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

jetenergy commented on Hivemind/Assets/Scripts/GameWorld.cs:
```diff
@@ -278,4 +303,41 @@ private IEnumerator LoadEnumerator(string name)
+        {
+            if (b.TeamID != teamId)
+            {
+                return b;
```
```suggestion
        foreach (BaseController controller in BaseControllerList)
        {
            if (controller.TeamID != teamId)
            {
                return controller;
```

ErikLouwerse commented on Hivemind/Assets/Editor/DataEditorDrawer.cs:
```diff
@@ -2,64 +2,69 @@
-	float GatheringPropCount = 4;
-	float CombatPropCount = 1;
+    float GatheringPropCount = 4;
+    float CombatPropCount = 2;
```
Can't these be private?

ErikLouwerse commented on Hivemind/Assets/Scripts/BaseBuilding/BuildingQueue.cs:
```diff
@@ -2,6 +2,7 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading.Tasks;
```
```suggestion

```

ErikLouwerse commented on Hivemind/Assets/Scripts/BaseBuilding/InBaseTrigger.cs:
```diff
@@ -17,20 +17,33 @@ public class InBaseTrigger : MonoBehaviour
+        if (ant != null)
         {
-            ant.isAtBase = true;
+            ant.currentSpeed  = ant.baseSpeed * 1.8f;
```
```suggestion
            ant.currentSpeed = ant.baseSpeed * 1.8f;
```

jetenergy committed:
converted scoutDeciSeconds to int because it doesnt use the properties of float

jetenergy commented on Hivemind/Assets/Scripts/SpatialPartition.cs:
```diff
@@ -28,7 +31,15 @@ void Start()
                 go.transform.Find("SpatialPartitioning");
                 go.GetComponent<SpatialPartitioning>().width = j;
                 go.GetComponent<SpatialPartitioning>().height = i;
+                go.GetComponent<SpatialPartitioning>().Id = id;
+                Grid.Add(id, go.GetComponent<SpatialPartitioning>());
```
```suggestion
                SpatialPartitioning sp = go.GetComponent<SpatialPartitioning>();
                sp.width = j;
                sp.height = i;
                sp.Id = id;
                Grid.Add(id, sp);
```

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/CombatMind.cs:
```diff
@@ -1,98 +1,381 @@
     public MindData GetData()
     {
-        return new CombatData(minEstimatedDifference, prefferedHealth, ant, busy);
+        return new CombatData( ant, busy, state, leavingBase, nextState, enterBase, TeleporterEntrance, AttackingQueen, target, EngageRange, enteredEnemyBase);
```
```suggestion
        return new CombatData(ant, busy, state, leavingBase, nextState, enterBase, TeleporterEntrance, AttackingQueen, target, EngageRange, enteredEnemyBase);
```

ErikLouwerse commented on Hivemind/Assets/Scripts/UI/WorkerRoomButton.cs:
```diff
@@ -0,0 +1,35 @@
+﻿using Assets.Scripts;
+using System.Collections;
```
```suggestion

```

ErikLouwerse commented on Hivemind/Assets/Scripts/UI/WorkerRoomButton.cs:
```diff
@@ -0,0 +1,35 @@
+    private void UpdateText(object sender, AmountChangedEventArgs args)
+    {
+        myText.text = $"Worker Room \n(Costs 10 rocks) \n({GameWorld.UnitLimit - args.Amount} Remaining)";
+        if(args.Amount >= GameWorld.UnitLimit)
```
```suggestion
        if (args.Amount >= GameWorld.UnitLimit)
```

ErikLouwerse commented on Hivemind/Assets/Scripts/UI/WorkerRoomButton.cs:
```diff
@@ -0,0 +1,35 @@
+        {
+            myButton.interactable = false;
+        }
+        else if(myBaseController.GetGameResources().GetResourceAmounts()[ResourceType.Rock] >= 10)
```
```suggestion
        else if (myBaseController.GetGameResources().GetResourceAmounts()[ResourceType.Rock] >= 10)
```

ErikLouwerse commented on Hivemind/Assets/Scripts/DataEditor.cs:
```diff
@@ -24,10 +29,31 @@ public IMind GenerateMind()
+
             case MindType.Combat:
-                return new CombatMind();
+                return new CombatMind(VisionRadius, AttackingQueen);
```
```suggestion
                return new CombatMind(VisionRadius, AttackingQueen);
                
```

ErikLouwerse commented on Hivemind/Assets/Scripts/GameWorld.cs:
```diff
@@ -244,25 +263,31 @@ private IEnumerator LoadEnumerator(string name)
+            foreach (UnitController controller in UnitControllerList)
+            {
+                controller.SetData(saveObject.TeamMindGroupData.FirstOrDefault(data => data.TeamId == controller.TeamId).MindGroupDataList);
+                if(controller.TeamId == LocalTeamId)
```
```suggestion
                if (controller.TeamId == LocalTeamId)
```

MMCQ33 committed:
Removed combattests, removed a field

removed combattests was an empty file. Removed mindpoints field that was unused.

jetenergy commented on Hivemind/Assets/Scripts/UI/WorkerRoomButton.cs:
```diff
@@ -0,0 +1,35 @@
+        else if(myBaseController.GetGameResources().GetResourceAmounts()[ResourceType.Rock] >= 10)
+        {
+            myButton.interactable = true;
+        }
```
word deze wel op false gezet wanneer resources < 10?

ErikLouwerse replied:
niet in deze klasse, dat werd al gedaan

ErikLouwerse commented on Hivemind/Assets/Scripts/MindGroup.cs:
```diff
@@ -129,4 +151,28 @@ public void SetData(MindGroupData data, GameObject parent, GameObject UiObject)
         }
     }
+
+    List<Ant> antsGottenNewMinds = new List<Ant>();
```
```suggestion
    private List<Ant> antsGottenNewMinds = new List<Ant>();
```

MMCQ33 committed:
renamed c

ErikLouwerse commented on Hivemind/Assets/Scripts/MindGroupList.cs:
```diff
@@ -113,8 +113,34 @@ internal void UpdateLayout(UnitGroup unit)
+        return Total;
     }
 
+    public int GetTotalPossibleAnts()
```
```suggestion

    public int GetTotalPossibleAnts()
```

MMCQ33 committed:
Update Hivemind/Assets/Scripts/BaseBuilding/BuildingQueue.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/Minds/CombatMind.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/UI/WorkerRoomButton.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/UI/WorkerRoomButton.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/UI/WorkerRoomButton.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/GameWorld.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/DataEditor.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/MindGroupList.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
fixed space because i am a retard and approved this

MMCQ33 committed:
Update Hivemind/Assets/Scripts/GameWorld.cs

Co-authored-by: René Duivenvoorden <renegrapjas@hotmail.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/SpatialPartition.cs

Co-authored-by: René Duivenvoorden <renegrapjas@hotmail.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/MindGroup.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Scripts/BaseBuilding/InBaseTrigger.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
changed position of property

MMCQ33 committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

MMCQ33 committed:
Made propcounts private

ErikLouwerse committed:
Set property which failed to save

ErikLouwerse committed:
Merge branch '#8-Combat' of https://github.com/zoompower/Hivemind into #8-Combat

jetenergy committed:
Added potential fix for job open when base ant dies
